test/Grpc.Net.Client.Tests/DiagnosticsTests.cs

```diff
@@ -0,0 +1,218 @@
+                response.TrailingHeaders.Add(GrpcProtocolConstants.MessageTrailer, "value");
+                return response;
+            });
+            var invoker = HttpClientCallInvokerFactory.Create(httpClient);
+
+            // Act & Assert
+            var a = new Activity("a").Start();
+            Assert.AreEqual("a", Activity.Current.OperationName);
+
+            var call = await Task.Run(() =>
+            {
+                var c = invoker.AsyncDuplexStreamingCall<HelloRequest, HelloReply>(ClientTestHelpers.ServiceMethod, string.Empty, new CallOptions());
+                Assert.AreEqual("a", Activity.Current.OperationName);
+
+                return c;
+            });
+            Assert.AreEqual("a", Activity.Current.OperationName);
+
+            var b = new Activity("b").Start();
+            Assert.AreEqual("b", Activity.Current.OperationName);
+
+            call.Dispose();
+            Assert.AreEqual("b", Activity.Current.OperationName);
+        }
+
+        [Test]
+        public void DiagnosticListener_MakeCall_ActivityWritten()
+        {
+            // Arrange
+            HttpRequestMessage? requestMessage = null;
+            HttpResponseMessage? responseMessage = null;
+            var httpClient = ClientTestHelpers.CreateTestClient(async request =>
+            {
+                requestMessage = request;
+
+                var streamContent = await ClientTestHelpers.CreateResponseContent(new HelloReply()).DefaultTimeout();
+                responseMessage = ResponseUtils.CreateResponse(HttpStatusCode.OK, streamContent, grpcStatusCode: StatusCode.Aborted);
+                responseMessage.TrailingHeaders.Add(GrpcProtocolConstants.MessageTrailer, "value");
+                return responseMessage;
+            });
+            var invoker = HttpClientCallInvokerFactory.Create(httpClient);
+
+            var result = new List<KeyValuePair<string, object>>();
+
+            // Act
+            using (GrpcDiagnostics.DiagnosticListener.Subscribe(new ObserverToList<KeyValuePair<string, object>>(result)))
+            {
+                var c = invoker.AsyncDuplexStreamingCall<HelloRequest, HelloReply>(ClientTestHelpers.ServiceMethod, string.Empty, new CallOptions());
+                c.Dispose();
+            }
+
+            // Assert
+            Assert.AreEqual(2, result.Count);
+            Assert.AreEqual(GrpcDiagnostics.ActivityStartKey, result[0].Key);
+            Assert.AreEqual(requestMessage, GetValueFromAnonymousType<HttpRequestMessage>(result[0].Value, "Request"));
+            Assert.AreEqual(GrpcDiagnostics.ActivityStopKey, result[1].Key);
+            Assert.AreEqual(requestMessage, GetValueFromAnonymousType<HttpRequestMessage>(result[1].Value, "Request"));
+            Assert.AreEqual(responseMessage, GetValueFromAnonymousType<HttpResponseMessage>(result[1].Value, "Response"));
+        }
+
+        [Test]
+        public void DiagnosticListener_MakeCall_ActivityHasNameAndDuration()
```

Comment:
Hmm looks like some of the logic is shared across this test and the previous. Maybe combine them?